resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreakerRegistry.java

```diff
@@ -261,4 +263,60 @@ CircuitBreaker circuitBreaker(String name,
         Supplier<CircuitBreakerConfig> circuitBreakerConfigSupplier,
         io.vavr.collection.Map<String, String> tags);
 
+    static Builder custom() {
+        return new Builder();
+    }
+
+    class Builder {
+
+        private static final String DEFAULT_CONFIG = "default";
+        private RegistryStore registryStore;
+        private Map<String, CircuitBreakerConfig> circuitBreakerConfigsMap;
+        private List<RegistryEventConsumer<CircuitBreaker>> registryEventConsumers;
+        private io.vavr.collection.Map<String, String> tags;
+
+        public Builder() {
+            this.circuitBreakerConfigsMap = new java.util.HashMap<>();
+            this.registryEventConsumers = new ArrayList<>();
+        }
+
+        public Builder withRegistryStore(RegistryStore registryStore) {
+            this.registryStore = registryStore;
+            return this;
+        }
+
+        public Builder withCircuitBreakerConfig(CircuitBreakerConfig circuitBreakerConfig) {
+            circuitBreakerConfigsMap.put(DEFAULT_CONFIG, circuitBreakerConfig);
+            return this;
+        }
+
+        public Builder withCircuitBreakerConfigDefaults() {
+            circuitBreakerConfigsMap.put(DEFAULT_CONFIG, CircuitBreakerConfig.ofDefaults());
```

Comment:
`withCircuitBreakerConfigDefaults` is not necessary.  If the config map is empty, the InMemoryCircuitBreakerRegistry creates a default config.

Reply:
@RobWin  I agree that if the config map is empty, InMemoryCircuitBreakerRegistry creates a default config but I implemented this method so that user can take conscious decision to create CircuitBreakerRegistry configs by default and does not leave the hashmap empty unknowingly. Let me know if that makes sense, I will remove the method otherwise.

Reply:
A CircuitBreakerRegistry always has a default CircuitBreakerConfig. It doesn't really matter if the user invokes `withCircuitBreakerConfigDefaults` or not. 
If this method exists, it might be confusing for a users way a CircuitBreakerRegistry has defaults, even if the user didn't invoke the method. I think we should remove it.